### DISCUSSION TITLE
How to add translation on Gitbook?

### DISCUSSION THREAD

**Discussion by mendax1234** [GitbookIO/community#225] [Feature Requests]
### What's your feature/idea?

l want to know whether l can add translation on my Gitbook?

### What's your desired outcome?

l think Gitbook will provide a button on the website and if l click the button, l can choose which language the content will be shown. And l hope that app.gitbook.com can also add a new section to translate the docs.

### What's the impact of this for you?

- [ ] Critical
- [X] High
- [ ] Medium
- [ ] Low

**Comment by addisonschultz** [comment]
Hi @mendax1234 ! Great feature request - this isn't something that is possible natively in the app, but is possible through our integration platform! I wrote a guide on this topic here! https://developer.gitbook.com/getting-started/guides/use-github-actions-to-translate-gitbook-pages

**Reply by mendax1234** [reply]
> Hi @mendax1234 ! Great feature request - this isn't something that is possible natively in the app, but is possible through our integration platform! I wrote a guide on this topic here! https://developer.gitbook.com/getting-started/guides/use-github-actions-to-translate-gitbook-pages

Thanks for your reply! l will have a look at it!

**Reply by mendax1234** [reply]
> Hi @mendax1234 ! Great feature request - this isn't something that is possible natively in the app, but is possible through our integration platform! I wrote a guide on this topic here! https://developer.gitbook.com/getting-started/guides/use-github-actions-to-translate-gitbook-pages

Hi! After reading your guide, I'd like to know if there is a specific example.  And is there any public repo on Github which has already realized this function? Thank you very much!

**Reply by addisonschultz** [reply]
Hey @mendax1234

I don't think I've seen any specific examples - But for example, you could consider using a GitHub action like this: https://github.com/3ru/gpt-translate

**Reply by mendax1234** [reply]
> Hey @mendax1234
> 
> I don't think I've seen any specific examples - But for example, you could consider using a GitHub action like this: https://github.com/3ru/gpt-translate

Thanks for your reply!

**Reply by mendax1234** [reply]
> Hey @mendax1234
> 
> I don't think I've seen any specific examples - But for example, you could consider using a GitHub action like this: https://github.com/3ru/gpt-translate

Hi @addisonschultz,

l have successfully integrated translation on my gitbook, but l still have some problems. Every time l finish writing in one language and then l want to translate it into another language, I find that l need to open a new tab to see the content in the original language. l think it is inconvenient for people to translate. And l think that every time l finish writing content, the gitbook or GitHub can copy them directly to the other language's folder, this may be an easy solution. Perhaps using Github Action can help me do this. l'd like to know if gitbook has any other better solution.

**Reply by addisonschultz** [reply]
Hi @mendax1234! Did you manage to use the GitHub action linked above? I'm a little confused at what you mean when you say you need to open a new tab to see the translations - are you doing this as a manual step?

**Reply by mendax1234** [reply]
> Hi @mendax1234! Did you manage to use the GitHub action linked above? I'm a little confused at what you mean when you say you need to open a new tab to see the translations - are you doing this as a manual step?

Hi @addisonschultz,

> are you doing this as a manual step?

Yes, l am doing this as a manual step.:joy: Besides, l find that the GitHub Action link above may use an OpenAI API, which is not free. Is there any free tool that also has been integrated into GitHub Action? 

Thanks!

**Reply by mendax1234** [reply]
l just found that l can create an OpenAi API key and l am trying with this GitHub action. 😂

**Reply by addisonschultz** [reply]
Yes - that is the route you should go. Unfortunately, I don't have any recommendations for free integrations that do translations

**Comment by mendax1234** [comment]
Thanks @addisonschultz l think this issue has been solved though there are still many deficiencies of this GPT-translation GitHub action.
